tests/tracer/test_writer.py

```diff
@@ -452,3 +452,19 @@ def test_flush_queue_raise():
     with pytest.raises(error):
         writer.write([])
         writer.flush_queue(raise_exc=True)
+
+
+def test_double_stop():
+    writer = AgentWriter(agent_url="http://dne:1234")
+    writer.write([])
+    assert writer.started
+    writer.stop()
+    assert writer.started
+    assert not writer.is_alive()
+    writer.stop()
+    assert writer.started
+    assert not writer.is_alive()
+
+    # Make sure it can start again.
+    writer.write([])
+    assert writer.started
```

Comment:
We should check that it's back alive:

```suggestion
    assert writer.started
    assert writer.is_alive()
```

Reply:
Strange, this actually fails the test 🤔

Reply:
Turns out the `started` flag doesn't actually get reset and for good reason:

```python
        if self._started.is_set():
>           raise RuntimeError("threads can only be started once")
E           RuntimeError: threads can only be started once
```

Reply:
I'll modify the test to assert the current behaviour.

Reply:
I opened up https://github.com/DataDog/dd-trace-py/pull/2125 to deal with this behaviour.